app/models/ont/library_factory.rb

```diff
@@ -1,183 +1,99 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/ClassLength
-
 # Ont namespace
 module Ont
   # LibraryFactory
   # The factory will create libraries in tubes from a given plate
   class LibraryFactory
     include ActiveModel::Model
 
-    validate :check_validation_errors,
-             :check_libraries,
-             :check_library_requests,
-             :check_tubes,
-             :check_container_materials
+    validate :check_library,
+             :check_tube,
+             :check_container_material
 
     def initialize(attributes = {})
-      @validation_errors = []
-      @libraries = []
-      @library_requests = []
-      @tubes = []
-      @container_materials = []
-
-      return unless fetch_and_validate_entities(attributes)
-
-      num_libraries.times do |lib_idx|
-        build_library(lib_idx, @tag_set.tags.count)
-      end
+      build_library(attributes)
     end
 
-    attr_reader :tubes
+    attr_reader :tube
 
     def save(**options)
       return false unless options[:validate] == false || valid?
 
-      @libraries.each { |lib| lib.save(validate: false) }
-      @library_requests.each { |lib_req| lib_req.save(validate: false) }
-      @tubes.each { |tube| tube.save(validate: false) }
-      @container_materials.each { |cont_mat| cont_mat.save(validate: false) }
+      @library.save(validate: false)
+      @tube.save(validate: false)
```

Comment:
We have a getter for all these instance variables so we should drop the `@` when only getting the variable.

Reply:
I know I've flip flopped on this decision in the past, but I read up GPGs on this and you should use the non-@ version and use getters and/or setters instead via `attr_reader`, `attr_writer` and `attr_accessor`, scoping those appropriately (like you already have)